### PR TITLE
feat: add `useReportPublication` hook (T-2513)

### PR DESCRIPTION
**Pull request by krzysu**
add `useReportPublication` hook

documentation https://docs.lens.xyz/docs/use-report-publication

### CODE REVIEW

File: packages/react/src/publication/useReaction.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export function useReaction({ profileId }: UseReactionArgs) {
   const addReaction = useCallback(
     async (args: ReactionArgs) => {
       setIsPending(true);
+      setError(null);
```
Comment (desfero) [inline packages/react/src/publication/useReaction.ts:26]: 👍

File: packages/react/src/publication/adapters/ReportPublicationGateway.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+          reason: PublicationReportingReason.Spam,
+          subreason: PublicationReportingSpamSubreason.SomethingElse,
+        },
+      };
+    default:
+      assertNever(reason, `Unknown report type`);
+  }
+};
+
+export class ReportPublicationGateway implements IReportPublicationGateway {
```
Comment (desfero) [inline packages/react/src/publication/adapters/ReportPublicationGateway.ts:149]: Would be good to have tests for the gateway

File: packages/react/src/publication/adapters/ReportPublicationPresenter.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import {
+  AlreadyReportedError,
+  IReportPublicationPresenter,
+} from '@lens-protocol/domain/use-cases/publications';
+import { Deferred, Result } from '@lens-protocol/shared-kernel';
+
+export class ReportPublicationPresenter implements IReportPublicationPresenter {
```
Comment (desfero) [inline packages/react/src/publication/adapters/ReportPublicationPresenter.ts:7]: Could we reuse our `PromiseResultPresenter` presenter and avoid having `ReportPublicationPresenter` at all?

**Review by desfero — CHANGES_REQUESTED** [review]

File: .changeset/calm-rockets-sit.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+"@lens-protocol/api-bindings": minor
+"@lens-protocol/domain": minor
```
[the remark this answers is not in the document]
Comment (krzysu) [reply .changeset/calm-rockets-sit.md:3]: thanks for the explanation! I will fix it

**Review by desfero — APPROVED** [review]

**Pull request merged**